pyiron_workflow/channels.py

```diff
@@ -777,7 +783,42 @@ def __init__(
                 object.
         """
         super().__init__(label=label, node=node)
-        self.callback: callable = callback
+        if self._is_node_method(callback) and self._takes_zero_arguments(callback):
+            self._callback: str = callback.__name__
+        else:
+            raise BadCallbackError(
+                f"The channel {self.label} on {self.node.label} got an unexpected "
+                f"callback: {callback}. "
+                f"Lives on node: {self._is_node_method(callback)}; "
+                f"take no args: {self._takes_zero_arguments(callback)} "
+            )
+
+    def _is_node_method(self, callback):
+        try:
+            return callback == getattr(self.node, callback.__name__)
+        except AttributeError:
+            return False
+
+    def _takes_zero_arguments(self, callback):
+        return callable(callback) and self._no_positional_args(callback)
+
+    @staticmethod
+    def _no_positional_args(func):
+        return (
+            sum(
+                1
+                for parameter in inspect.signature(func).parameters.values()
+                if (
+                    parameter.default == inspect.Parameter.empty
+                    and parameter.kind != inspect._ParameterKind.VAR_KEYWORD
+                )
+            )
+            == 0
+        )
```

Comment:
```suggestion
        return all([
            parameter.default != inspect.Parameter.empty
            or parameter.kind == inspect.Parameter.VAR_KEYWORD
            for parameter in inspect.signature(func).parameters.values()
        ])
```

Reply:
I changed it silently, but apparently `inspect._ParameterKind.VAR_KEYWORD` is the same as `inspect.Parameter.VAR_KEYWORD`